drivers/wifi/esp32/src/esp_wifi_drv.c

```diff
@@ -44,6 +56,22 @@ K_MSGQ_DEFINE(esp_wifi_msgq, sizeof(system_event_t), 10, 4);
 K_THREAD_STACK_DEFINE(esp_wifi_event_stack, CONFIG_ESP32_WIFI_EVENT_TASK_STACK_SIZE);
 static struct k_thread esp_wifi_event_thread;
 
+static struct net_mgmt_event_callback esp32_dhcp_cb;
+
+static void wifi_event_handler(struct net_mgmt_event_callback *cb, uint32_t mgmt_event,
+			       struct net_if *iface)
+{
+	const struct wifi_status *status = (const struct wifi_status *)cb->info;
+
+	switch (mgmt_event) {
+	case NET_EVENT_IPV4_DHCP_BOUND:
+		wifi_mgmt_raise_connect_result_event(esp32_wifi_iface, 0);
```

Comment:
Typically, this is only for L2 connected, so, if we delay/fail this in case DHCP delays/fails, then applications behave differently with ESP32 vs. other Wi-Fi chips, is that okay? Ideally DHCP restart/start should be handled by stack/apps higher up. In `nrf700x` DHCP restart is handled by wpa_supplicant, FYI here also a new API was added https://github.com/zephyrproject-rtos/zephyr/pull/49152

Reply:
Sure, I get that. And this is only used when `CONFIG_ESP32_WIFI_STA_AUTO_DHCPV4` is selected. By default, DHCP should be handled by the app. Thanks for pointing the DHCP restar PR.